src/main/java/com/synopsys/integration/jira/common/cloud/service/IssueService.java

```diff
@@ -67,30 +70,80 @@ public IssueCreationResponseModel createIssue(IssueCreationRequestModel requestM
+                                                               .collect(Collectors.toList());
+        String issueTypeId = null;
+        for (IssueTypeResponseModel issueTypeCandidate : issueTypeCandidates) {
+            IssueTypeScope scope = issueTypeCandidate.getScope();
+            if (null != scope) {
+                IdComponent scopedProject = scope.getProject();
+                if (null != scopedProject) {
+                    issueTypeId = scopedProject.getId();
+                    // Exact match. Search complete.
+                    break;
+                }
+            } else {
+                // Possible match. There could be a better one.
+                issueTypeId = issueTypeCandidate.getId();
+            }
+        }
+
+        if (null == issueTypeId) {
+            throw new JiraPreconditionNotMetException(String.format("Issue type not found; issue type %s", issueTypeName));
+        }
 
         IssueRequestModelFieldsBuilder fieldsBuilder = new IssueRequestModelFieldsBuilder();
         fieldsBuilder.copyFields(requestModel.getFieldsBuilder());
 
-        fieldsBuilder.setIssueType(foundIssueType.getId());
+        fieldsBuilder.setIssueType(issueTypeId);
         if (StringUtils.isNotBlank(reporterEmail)) {
             String accountId = retrieveUserAccountId(reporterEmail);
             fieldsBuilder.setReporterId(accountId);
         }
-        fieldsBuilder.setProject(foundProject.getId());
+        fieldsBuilder.setProject(projectId);
 
         Map<String, List<FieldUpdateOperationComponent>> update = new HashMap<>();
         IssueRequestModel issueRequestModel = new IssueRequestModel(fieldsBuilder, update, requestModel.getProperties());
         return createIssue(issueRequestModel);
     }
 
+    private String retrieveIssueTypeId(String issueTypeName, String projectId) throws IntegrationException {
+        List<IssueTypeResponseModel> allIssueTypes = issueTypeService.getAllIssueTypes();
+        List<IssueTypeResponseModel> issueTypeCandidates = allIssueTypes
+                                                               .stream()
+                                                               .filter(issueType -> issueType.getName().equalsIgnoreCase(issueTypeName))
+                                                               .collect(Collectors.toList());
+        String issueTypeId = null;
+        for (IssueTypeResponseModel issueTypeCandidate : issueTypeCandidates) {
+            IssueTypeScope scope = issueTypeCandidate.getScope();
+            if (null != scope) {
+                IdComponent scopedProject = scope.getProject();
+                if (null != scopedProject) {
+                    // Exact match. Search complete.
+                    return scopedProject.getId();
+                }
+            } else {
+                // Possible match. There could still be a better candidate.
+                issueTypeId = issueTypeCandidate.getId();
+            }
+        }
```

Comment:
Is this not the same code as above? Should the above code just call this method?

Reply:
I fixed it.